resources/build/build-utils.sh

```diff
@@ -35,17 +35,17 @@
 SHLVL=0
 
 function findKeymanRoot() {
-    # See https://stackoverflow.com/questions/59895/how-to-get-the-source-directory-of-a-bash-script-from-within-the-script-itself
-    # None of the answers are 100% correct for cross-platform
-    # On macOS, requires coreutils (`brew install coreutils`)
-    local SCRIPT=$(readlink -f "${BASH_SOURCE[0]}")
-    KEYMAN_ROOT="${SCRIPT%/*/*/*}"
-    readonly KEYMAN_ROOT
+  # We don't need readlink here because our standard script prolog does a
+  # readlink -f already so we will have already escaped from any symlinks
+  # But we still need to canonicalize paths to remove ../../..
+  KEYMAN_ROOT="${BASH_SOURCE[0]%/*/*/*}"
+  KEYMAN_ROOT="$( cd "$KEYMAN_ROOT" && echo "$PWD" )"
+  readonly KEYMAN_ROOT
```

Comment:
Regarding one of the [other comment chains](https://github.com/keymanapp/keyman/pull/8416#discussion_r1134876300):  I take it that this block interacts with the `..`-based relative pathing in this folder's scripts to help optimize startup time for scripts under the `resources` folder?

I do have to admit that I can't quite follow line 41.  My best guess up until now (from the previous, single-case uses) is that each `/*` cuts off the last path component?  But I feel that guess can't be correct, hence my asking for clarity.

Reply:
Yes, each `/*` matches a path component at the end of the string ([the `%` token](https://tldp.org/LDP/abs/html/string-manipulation.html)) and cuts it off.

Reply:
So... how does this work with `${KEYMAN_ROOT}/web/build.sh`?  As in, how do we keep it from cutting off too much?

Reply:
Oh right, `BASH_SOURCE`, not `THIS_SCRIPT`.

Reply:
Because the `${BASH_SOURCE[0]}` that we'll have in that case at this point will be `$THIS_SCRIPT_PATH/../resources/build/build-utils.sh` which resolves to `some-path.../web/../resources/build/build-utils.sh` and so we chop `/resources/build/build-utils.sh` off (matching `/*/*/*`), giving us `some-path.../web/..`, which we then use `cd`/`pwd` to resolve.

But as I said, I may roll this change back because of its inherent fragility.